os/mm/mm_heap/mm_initialize.c

```diff
@@ -263,7 +263,8 @@ int mm_initialize(FAR struct mm_heap_s *heap, FAR void *heapstart, size_t heapsi
 	for (i = 0; i < CONFIG_MAX_TASKS; i++) {
 		heap->alloc_list[i].pid = HEAPINFO_INIT_INFO;
 	}
-	heap->total_alloc_size = heap->peak_alloc_size = 0;
+	/* two gaurd nodes are always allocated for the start and end of the heap */
+	heap->total_alloc_size = heap->peak_alloc_size = 2 * SIZEOF_MM_ALLOCNODE;
```

Comment:
we can have multiple heap regions
Could you check if it should be `heap->peak_alloc_size = 2 * SIZEOF_MM_ALLOCNODE * CONFIG_KMM_REGIONS`

```
struct mm_heap_s {
        ... 
	FAR struct mm_allocnode_s *mm_heapstart[CONFIG_KMM_REGIONS];
	FAR struct mm_allocnode_s *mm_heapend[CONFIG_KMM_REGIONS];
        ....
```

Reply:
Thanks for pointing it out. However also have USER heap along with KERNEL heap. So, to support them both, I have moved the logic to mm_addregion. Please review.